datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/glossary/GlossaryUtilsTest.java

```diff
@@ -1,30 +1,102 @@
+    Mockito.when(mockContext.getAuthentication()).thenReturn(Mockito.mock(Authentication.class));
+
+    GlossaryNodeInfo parentNode1 = new GlossaryNodeInfo().setParentNode(GlossaryNodeUrn.createFromString(
+      "urn:li:glossaryNode:parent_node2")
+    );
+    GlossaryNodeInfo parentNode2 = new GlossaryNodeInfo().setParentNode(GlossaryNodeUrn.createFromString(
+        "urn:li:glossaryNode:parent_node3")
+    );
+  
+    GlossaryNodeInfo parentNode3 = new GlossaryNodeInfo();
+  
+    Map<String, EnvelopedAspect> parentNode1Aspects = new HashMap<>();
+    parentNode1Aspects.put(GLOSSARY_NODE_INFO_ASPECT_NAME, new EnvelopedAspect().setValue(new Aspect(
+        new GlossaryNodeInfo().setDefinition("node parent 1").setParentNode(parentNode1.getParentNode()).data()
+    )));
+  
+    Map<String, EnvelopedAspect> parentNode2Aspects = new HashMap<>();
+    parentNode2Aspects.put(GLOSSARY_NODE_INFO_ASPECT_NAME, new EnvelopedAspect().setValue(new Aspect(
+        new GlossaryNodeInfo().setDefinition("node parent 2").setParentNode(parentNode2.getParentNode()).data()
+    )));
+
+    Map<String, EnvelopedAspect> parentNode3Aspects = new HashMap<>();
+    parentNode3Aspects.put(GLOSSARY_NODE_INFO_ASPECT_NAME, new EnvelopedAspect().setValue(new Aspect(
+        new GlossaryNodeInfo().setDefinition("node parent 3").data()
+    )));
+
+    Mockito.when(mockClient.getV2(
+      Mockito.eq(Constants.GLOSSARY_NODE_ENTITY_NAME),
+      Mockito.eq(parentNodeUrn1),
+      Mockito.eq(ImmutableSet.of(GLOSSARY_NODE_INFO_ASPECT_NAME)),
+      Mockito.any(Authentication.class)
+    )).thenReturn(new EntityResponse().setAspects(new EnvelopedAspectMap(parentNode1Aspects)));
+
+    Mockito.when(mockClient.getV2(
+      Mockito.eq(Constants.GLOSSARY_NODE_ENTITY_NAME),
+      Mockito.eq(parentNodeUrn2),
+      Mockito.eq(ImmutableSet.of(GLOSSARY_NODE_INFO_ASPECT_NAME)),
+      Mockito.any(Authentication.class)
+    )).thenReturn(new EntityResponse().setAspects(new EnvelopedAspectMap(parentNode2Aspects)));
+
+    Mockito.when(mockClient.getV2(
+      Mockito.eq(Constants.GLOSSARY_NODE_ENTITY_NAME),
+      Mockito.eq(parentNodeUrn3),
+      Mockito.eq(ImmutableSet.of(GLOSSARY_NODE_INFO_ASPECT_NAME)),
+      Mockito.any(Authentication.class)
+    )).thenReturn(new EntityResponse().setAspects(new EnvelopedAspectMap(parentNode3Aspects)));
+
+    final ResourceSpec resourceSpec3 = new ResourceSpec(parentNodeUrn.getEntityType(), parentNodeUrn3.toString());
+    mockAuthRequest("MANAGE_GLOSSARY_CHILDREN", AuthorizationResult.Type.DENY, resourceSpec3);
+
+    final ResourceSpec resourceSpec2 = new ResourceSpec(parentNodeUrn.getEntityType(), parentNodeUrn2.toString());
+    mockAuthRequest("MANAGE_GLOSSARY_CHILDREN", AuthorizationResult.Type.DENY, resourceSpec2);
+
+    final ResourceSpec resourceSpec1 = new ResourceSpec(parentNodeUrn.getEntityType(), parentNodeUrn1.toString());
+    mockAuthRequest("MANAGE_GLOSSARY_CHILDREN", AuthorizationResult.Type.DENY, resourceSpec1);
```

Comment:
minor: is this part necessary since you create these 3 specs and mock them in the tests below as well?

Reply:
The test cases have the MANAGE_ALL_GLOSSARY_CHILDREN privilege and the setupTests have the MANAGE_GLOSSARY_CHILDREN privilege.

Reply:
ohh gotcha makes sense